src/Mod/PartDesign/Gui/TaskHelixParameters.cpp

```diff
@@ -0,0 +1,520 @@
+    connect(ui->inputMode, SIGNAL(activated(int)),
+            this, SLOT(onModeChanged(int)));
+
+
+    this->groupLayout()->addWidget(proxy);
+
+    // Temporarily prevent unnecessary feature recomputes
+    ui->axis->blockSignals(true);
+    ui->pitch->blockSignals(true);
+    ui->height->blockSignals(true);
+    ui->turns->blockSignals(true);
+    ui->coneAngle->blockSignals(true);
+    ui->checkBoxLeftHanded->blockSignals(true);
+    ui->checkBoxReversed->blockSignals(true);
+
+
+    //bind property mirrors
+    PartDesign::ProfileBased* pcFeat = static_cast<PartDesign::ProfileBased*>(vp->getObject());
+
+    PartDesign::Helix* rev = static_cast<PartDesign::Helix*>(vp->getObject());
+
+    this->propAngle = &(rev->Angle);
+    this->propPitch = &(rev->Pitch);
+    this->propHeight = &(rev->Height);
+    this->propTurns = &(rev->Turns);
+    this->propReferenceAxis = &(rev->ReferenceAxis);
+    this->propLeftHanded = &(rev->LeftHanded);
+    this->propReversed = &(rev->Reversed);
+    this->propMode = &(rev->Mode);
+
+
+    double pitch = propPitch->getValue();
+    double height = propHeight->getValue();
+    double turns = propTurns->getValue();
+    double angle = propAngle->getValue();
+    bool leftHanded = propLeftHanded->getValue();
+    bool reversed = propReversed->getValue();
+    int index = propMode->getValue();
+
+    ui->pitch->setValue(pitch);
+    ui->height->setValue(height);
+    ui->turns->setValue(turns);
+    ui->coneAngle->setValue(angle);
+    ui->checkBoxLeftHanded->setChecked(leftHanded);
+    ui->checkBoxReversed->setChecked(reversed);
+    ui->inputMode->setCurrentIndex(index);
+
+    blockUpdate = false;
+    updateUI();
+
+    // enable use of parametric expressions for the numerical fields
+    ui->pitch->bind(static_cast<PartDesign::Helix *>(pcFeat)->Pitch);
+    ui->height->bind(static_cast<PartDesign::Helix *>(pcFeat)->Height);
+    ui->turns->bind(static_cast<PartDesign::Helix *>(pcFeat)->Turns);
+    ui->coneAngle->bind(static_cast<PartDesign::Helix *>(pcFeat)->Angle);
+
+    ui->axis->blockSignals(false);
+    ui->pitch->blockSignals(false);
+    ui->height->blockSignals(false);
+    ui->turns->blockSignals(false);
+    ui->coneAngle->blockSignals(false);
+    ui->checkBoxLeftHanded->blockSignals(false);
+    ui->checkBoxReversed->blockSignals(false);
+
+    setFocus ();
+
+    //show the parts coordinate system axis for selection
+    PartDesign::Body * body = PartDesign::Body::findBodyOf ( vp->getObject () );
+    if(body) {
+        try {
+            App::Origin *origin = body->getOrigin();
+            ViewProviderOrigin* vpOrigin;
+            vpOrigin = static_cast<ViewProviderOrigin*>(Gui::Application::Instance->getViewProvider(origin));
+            vpOrigin->setTemporaryVisibility(true, false);
+        } catch (const Base::Exception &ex) {
+            ex.ReportException();
+        }
+     }
+}
+
+void TaskHelixParameters::fillAxisCombo(bool forceRefill)
```

Comment:
Would it be possible here to omit or disable the invalid axes, or is that something that's too difficult to determine at this point?

Reply:
I will take a look at how to improve this.